flink-streaming-connectors/flink-connector-kafka-base/src/main/java/org/apache/flink/streaming/connectors/kafka/internals/metrics/AvgKafkaMetricAccumulator.java

```diff
@@ -127,12 +127,17 @@ private static AvgSumCount getAvgSumCount(Avg avg) {
 	}
 
 	private void writeObject(ObjectOutputStream out) throws IOException {
-		Measurable thisMeasurable = DefaultKafkaMetricAccumulator.getMeasurableFromKafkaMetric(this.kafkaMetric);
-		if(!(thisMeasurable instanceof Avg) ) {
-			throw new RuntimeException("Must be of type Avg");
+		try {
+			Measurable thisMeasurable = DefaultKafkaMetricAccumulator.getMeasurableFromKafkaMetric(this.kafkaMetric);
+			if(!(thisMeasurable instanceof Avg) ) {
+				throw new RuntimeException("Must be of type Avg");
+			}
+			this.lastSumCount = getAvgSumCount((Avg) thisMeasurable);
+			out.defaultWriteObject();
+		} catch(Exception e) {
```

Comment:
A `RuntimeException` is also an `Exception`. Do you really want to catch this exception in the catch clause?